contracts/contracts/chains/ethereum/EthereumMessengers.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "OpenZeppelin/openzeppelin-contracts@4.8.0/contracts/access/Ownable.sol";
+
+import "../../../interfaces/IMessenger.sol";
+import "../../RestrictedCalls.sol";
+import "../../Resolver.sol";
+
+contract EthereumL1Messenger is IMessenger, RestrictedCalls {
+    function callAllowed(address, address) external pure returns (bool) {
+        // The call must be delivered by the Resolver
+        // Since we are on the same chain as the Resolver
+        // we should never see an execution of this function.
+        revert("Unexpected call to `callAllowed`.");
+    }
+
+    // Simply forward the message to the target on L1
+    function sendMessage(address target, bytes calldata message)
+        external
+        restricted(block.chainid)
+    {
+        (bool sent, ) = target.call(message);
+        require(sent, "sending message failed.");
```

Comment:
Nitpick: I would omit full stop at the end, to be consistent with the rest of our messages.